src/main/java/nl/tudelft/cse1110/andy/grade/GradeCalculator.java

```diff
@@ -24,6 +24,11 @@ public int calculateFinalGrade(GradeValues gradeValues, GradeWeight weights) {
         if(finalGrade < 0 || finalGrade > 100)
             throw new RuntimeException("Invalid grade calculation");
 
+        // Grades between 99.5 and 100 should be rounded down to 99 instead of up
+        if (finalGrade == 100 && finalDecimalGrade < 1) {
```

Comment:
Can we use the sum of integers (with all the points from the different criteria) rather than the float? That’d be a tad safer against yet another rounding error.

Reply:
I changed it to compare the scores of all components with a positive grade instead of the decimal grade